math/dec.go

```diff
@@ -490,7 +490,8 @@ func (d LegacyDec) Power(power uint64) LegacyDec {
 
 func (d LegacyDec) PowerMut(power uint64) LegacyDec {
 	if power == 0 {
-		d.SetInt64(1)
+		// Set to 1 with the correct precision.
+		d.SetInt(precisionReuse)
```

Comment:
Just checked out the repo actually because the build was failing (this method does not exist).
However, looking at `SetInt64`, we can see it does:
```go
	d.i.SetInt64(i)
	d.i.Mul(d.i, precisionReuse)
	return d
```
So what is the precision issue?

Reply:
Meant to do:

`d.i.Set(i)` like here: 
https://github.com/osmosis-labs/cosmos-sdk/pull/451/commits/aeaa930dc49507dc64c92cb923307c720083b119

`SetInt64` allocates a new buffer under the hood

Reply:
I suspect `z.make(2)` is what allocates the new buffer. That does not update `d`'s buffer in `PowerMut`
```go
func (z nat) setUint64(x uint64) nat {
	// single-word value
	if w := Word(x); uint64(w) == x {
		return z.setWord(w)
	}
	// 2-word value
	z = z.make(2)
	z[1] = Word(x >> 32)
	z[0] = Word(x)
	return z
}
```

Reply:
Fix pushed: [721793c](https://github.com/cosmos/cosmos-sdk/pull/16266/commits/721793cecb6decce66cd061fbf07e14cf3d816b9)